builtin/providers/aws/config.go

```diff
@@ -195,23 +198,25 @@ func (c *Config) Client() (interface{}, error) {
 		dynamoSess := sess.Copy(&aws.Config{Endpoint: aws.String(c.DynamoDBEndpoint)})
 		kinesisSess := sess.Copy(&aws.Config{Endpoint: aws.String(c.KinesisEndpoint)})
 
-		// These two services need to be set up early so we can check on AccountID
-		client.iamconn = iam.New(awsIamSess)
-		client.stsconn = sts.New(sess)
+		if c.SkipIamValidation == false {
+			// These two services need to be set up early so we can check on AccountID
+			client.iamconn = iam.New(awsIamSess)
+			client.stsconn = sts.New(sess)
 
-		err = c.ValidateCredentials(client.stsconn)
-		if err != nil {
-			errs = append(errs, err)
-			return nil, &multierror.Error{Errors: errs}
-		}
-		accountId, err := GetAccountId(client.iamconn, client.stsconn, cp.ProviderName)
-		if err == nil {
-			client.accountid = accountId
-		}
+			err = c.ValidateCredentials(client.stsconn)
+			if err != nil {
+				errs = append(errs, err)
+				return nil, &multierror.Error{Errors: errs}
+			}
+			accountId, err := GetAccountId(client.iamconn, client.stsconn, cp.ProviderName)
+			if err == nil {
+				client.accountid = accountId
+			}
 
-		authErr := c.ValidateAccountId(client.accountid)
-		if authErr != nil {
-			errs = append(errs, authErr)
+			authErr := c.ValidateAccountId(client.accountid)
```

Comment:
I'm thinking that some people theoretically may have the metadata API still available, so they may still want to perform the account ID validation, but they may not want to use IAM/STS API at all.

Therefore in the following case
```hcl
provider "aws" {
  skip_iam_validation = true
  allowed_account_ids = ["1111111111"]
}
```
ID check will be silently ignored even though the Account ID may be available via Metadata API.

As you can see in the `ValidateAccountId` implementation, it only performs validation if `AllowedAccountIds` or `ForbiddenAccountIds` are non-empty, so it's totally safe to keep it outside of the conditional block.

Likewise `ValidateCredentials` would _gracefully degrade_ - i.e. return empty string if it was impossible to find account ID from all available sources.

That said I understand you may want to eliminate any extra API calls to APIs that you know are unavailable in your environment - so how about this?

```hcl
provider "aws" {
  skip_credentials_validation = true
  skip_account_id_validation = true
  skip_metadata_api_check = true
}
```
That way we can cater both groups of people - those that don't want Terraform to perform _any_ API calls and those who _only_ want to ignore IAM/STS, but still want to perform account ID validation.

We could also mark `skip_account_id_validation` as `ConflictsWith: []string{"allowed_account_ids", "forbidden_account_ids"}`, but it would fail early anyway, because `"" != "given-account-id"`.